packages/pluggableWidgets/datagrid-text-filter-web/src/components/FilterComponent.tsx

```diff
@@ -0,0 +1,63 @@
+import { createElement, Dispatch, ReactElement, useEffect, useState } from "react";
+import { FilterSelector } from "./FilterSelector";
+import { ObjectItem, ListAttributeValue } from "mendix";
+import { DefaultFilterEnum } from "../../typings/DatagridTextFilterProps";
+
+interface FilterComponentProps {
+    defaultFilter: DefaultFilterEnum;
+    value?: string;
+    placeholder?: string;
+    filterDispatcher: Dispatch<{ filter(item: ObjectItem, attribute: ListAttributeValue): boolean }>;
+}
+
+export function FilterComponent(props: FilterComponentProps): ReactElement {
+    const [type, setType] = useState<DefaultFilterEnum>(props.defaultFilter);
+    const [value, setValue] = useState(props.value);
+
+    useEffect(() => {
+        if (props.filterDispatcher) {
+            props.filterDispatcher({
+                filter: (item, attr): boolean => {
+                    if (!value) {
+                        return true;
+                    }
+                    const dataValue = attr(item).displayValue.toLowerCase();
+                    const filterValue = value.toLowerCase();
+                    switch (type) {
+                        case "contains":
+                            return dataValue.includes(filterValue);
+                        case "startsWith":
+                            return dataValue.startsWith(filterValue);
+                        case "endsWith":
+                            return dataValue.endsWith(filterValue);
+                        case "greater":
+                            return dataValue > filterValue;
```

Comment:
Does greater, smaller makes sense for text filters ?

Reply:
Yes, this is a requiremente for zipcodes for example.